[ENG-8721][docs] document new custom build functions (#22675)
expo/expo: docs/pages/preview/custom-build-config-schema.mdx

        run: ls assets
```

#### `eas/use_npm_token`

> Configures node package managers (npm / yarn / pnpm) for use with private packages, published either to npm or private registry.

Reviewer: ```suggestion
Configures node package managers (npm, pnpm, or Yarn) to use with private packages, published either to npm or private registry.
```

[External product casing](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#follow-external-product-casing)

> Set `NPM_TOKEN` in your projects secrets and this function will configure the build environment by creating `.npmrc` with the token.

Reviewer: ```suggestion
Set `NPM_TOKEN` in your project's secrets, and this function will configure the build environment by creating **.npmrc** with the token.
```

```yaml test.yml
build:
  name: Install private npm modules
  steps:
    - eas/checkout
    # @info #
    - eas/use_npm_token
    # @end #
    - run:
        name: Install dependencies
        run: npm install # <---- Can now install private packages
```

#### `eas/install_node_modules`

> Installs node modules using the package manager detected base on your project. Works with monorepos.

Reviewer: ```suggestion
Installs node modules using the package manager (npm, pnpm, or Yarn) detected based on your project. Works with monorepos.
```

> Supported package managers: npm, yarn, pnpm

Reviewer: ```suggestion
```

I think we can list the package managers above instead of here.
